chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2513 @@
+
+# DTMF
+GROUPCODE = ["","Off", "*", "#", "A", "B", "C", "D"]
+
+AB_LIST = ["A", "B"]
+ALMOD_LIST = ["Site", "Tone", "Code"]
+BANDWIDTH_LIST = ["Wide", "Narrow"]
+COLOR_LIST = ["Off", "Blue", "Orange", "Purple"]
+DTMFSPEED_LIST = ["%s ms" % x for x in range(50, 2010, 10)]
+DTMFST_LIST = ["OFF", "DT-ST", "ANI-ST", "DT+ANI"]
+MODE_LIST = ["Channel", "Name", "Frequency"]
+PONMSG_LIST = ["Full", "Message"]
+PTTID_LIST = ["Off", "BOT", "EOT", "Both"]
+PTTIDCODE_LIST = ["%s" % x for x in range(1, 16)]
+RTONE_LIST = ["1000 Hz", "1450 Hz", "1750 Hz", "2100Hz"]
+RESUME_LIST = ["TO", "CO", "SE"]
+ROGERRX_LIST = ["Off"] + AB_LIST
+RPSTE_LIST = ["OFF"] + ["%s" % x for x in range(1, 11)]
+SAVE_LIST = ["Off", "1:1", "1:2", "1:3", "1:4"]
+SCODE_LIST = ["%s" % x for x in range(1, 16)]
+SHIFTD_LIST = ["Off", "+", "-"]
+STEDELAY_LIST = ["OFF"] + ["%s ms" % x for x in range(100, 1100, 100)]
+STEPS = [2.5, 5.0, 6.25, 10.0, 12.5, 25.0]
+STEP_LIST = [str(x) for x in STEPS]
+STEPS = [2.5, 5.0, 6.25, 10.0, 12.5, 20.0, 25.0, 50.0]
+STEP291_LIST = [str(x) for x in STEPS]
+TDRAB_LIST = ["Off"] + AB_LIST
+TDRCH_LIST = ["CH%s" % x for x in range(1, 129)]
+TIMEOUT_LIST = ["%s sec" % x for x in range(15, 615, 15)] + \
+    ["Off (if supported by radio)"]
+TXPOWER_LIST = ["High", "Low"]
+TXPOWER3_LIST = ["High", "Mid", "Low"]
+VOICE_LIST = ["Off", "English", "Chinese"]
+VOX_LIST = ["OFF"] + ["%s" % x for x in range(1, 11)]
+WORKMODE_LIST = ["Frequency", "Channel"]
+
+SETTING_LISTS = {
+    "almod": ALMOD_LIST,
+    "aniid": PTTID_LIST,
+    "displayab": AB_LIST,
+    "dtmfst": DTMFST_LIST,
+    "dtmfspeed": DTMFSPEED_LIST,
+    "mdfa": MODE_LIST,
+    "mdfb": MODE_LIST,
+    "ponmsg": PONMSG_LIST,
+    "pttid": PTTID_LIST,
+    "rtone": RTONE_LIST,
+    "rogerrx": ROGERRX_LIST,
+    "rpste": RPSTE_LIST,
+    "rxled": COLOR_LIST,
+    "save": SAVE_LIST,
+    "scode": PTTIDCODE_LIST,
+    "screv": RESUME_LIST,
+    "sftd": SHIFTD_LIST,
+    "stedelay": STEDELAY_LIST,
+    "step": STEP_LIST,
+    "step291": STEP291_LIST,
+    "tdrab": TDRAB_LIST,
+    "tdrch": TDRCH_LIST,
+    "timeout": TIMEOUT_LIST,
+    "txled": COLOR_LIST,
+    "txpower": TXPOWER_LIST,
+    "txpower3": TXPOWER3_LIST,
+    "voice": VOICE_LIST,
+    "vox": VOX_LIST,
+    "widenarr": BANDWIDTH_LIST,
+    "workmode": WORKMODE_LIST,
+    "wtled": COLOR_LIST
+}
+
+GMRS_FREQS1 = [462.5625, 462.5875, 462.6125, 462.6375, 462.6625,
+               462.6875, 462.7125]
+GMRS_FREQS2 = [467.5625, 467.5875, 467.6125, 467.6375, 467.6625,
+               467.6875, 467.7125]
+GMRS_FREQS3 = [462.5500, 462.5750, 462.6000, 462.6250, 462.6500,
+               462.6750, 462.7000, 462.7250]
+GMRS_FREQS_B = GMRS_FREQS2 * 3 + GMRS_FREQS3 * 2
+
+GMRS_FREQ_A = GMRS_FREQS1 + GMRS_FREQS3
+
+HAM_FREQS = [162.55000, 162.40000, 162.47500, 162.42500, 162.45000,
+             162.50000, 162.52500, 161.65000, 161.77500, 161.75000]
+HAM_NAME = ["NOAA 1", "NOAA 2", "NOAA 3", "NOAA 4", "NOAA 5", "NOAA 6",
```

Comment:
That's right, in HAM mode, the receiving frequency in the 189-199 channel of H8 is not allowed to be modified (you are right), except for this wide and narrow band and scan addition, the 189-199 channel option cannot be modified , I don't know how to gray it out or block it so that it is not allowed to be modified.

Reply:
Okay, one thing you can do is make `validate_memory()` fail if they change fields on those memories. The old UI also had `immutable` fields on the memory, but I haven't implemented that in the new UI. I can try to work on that, which might be better for this case.